File: CSharp-01-Fundamentals/05. Conditional-Statements/Homework/05. Conditional-Statements/P01. Exchange If Greater/P01. Exchange If Greater.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
/********************************************************************************
    Exchange numbers

    Description

    Write a program that reads two double values from the console A and B, stores them in variables and exchanges their values if the first one is greater than the second one. Use an if-statement. As a result print the values of the variables A and B, separated by a space.

    Input

    On the first line, you will receive the value of A
    On the second line, you will receive the value of B
    Output

    On the only output line, print the values of the two variables, separated by a whitespace
    Constraints

    A and B will always be valid real numbers between -100 and 100
    Time limit: 0.1s
    Memory limit: 16MB
    Sample tests

Input	Output
5
3	3 5
2
4	2 4
3,3
3,3	3,3 3,3
     
********************************************************************************/

namespace P01.ExchangeIfGreater
{
    class ExchangeIfGreater
    {
        static void Main(string[] args)
        {
            double a = double.Parse(Console.ReadLine());
            double b = double.Parse(Console.ReadLine());

            if (a > b)
            {
                double aOld = a;
                a = b;
                b = aOld;
            }

           
            Console.WriteLine("{0} {1}", a, b);
        }
    }
}

File: My-Homeworks-ExamPreparations-And-Excersizes/JavaScript Fundamentals/05. Conditional Statements/Homework/P01. Exchange if greater/P01. Exchange if greater.js
/*
 Exchange if greater

 Description

 Write an  if  statement that takes two double variables  a  and  b  and exchanges their values if the first one is greater than the second. As a result print the values  a  and  b , separated by a space.

 Input
 •The input will consist of an array containing two values -  a  and  b  represented as strings

 Output
 •The output should be a single line containing two numbers

 Constraints
 •Time limit: 0.2s
 •Memory limit: 16MB

 Sample tests

 Input

 Output

 5
 2 2 5
 3
 4 3 4
 5.5
 4.5 4.5 5.5

 */


// var input = [5, 2, 3, 4, 5.5, 4.5];
// solve(input);

function solve(args) {

    // var argsLen = args.length;
    // for(var i = 1; i < argsLen; i = i + 2){
    //     var a = args[i - 1];
    //     var b = args[i];
    //
    //     exchangeIfGreater(a, b)
    // }

    var a = args[0];
    var b = args[1];

    exchangeIfGreater(a, b)

    function exchangeIfGreater(a, b) {
        a = parseFloat(a);
        b = parseFloat(b);
        diffComparer = 0.000000001;

        var isGreaterA = (a - b > diffComparer);
        if (isGreaterA) {
            var temp = a;
            a = b;
            b = temp;
        }

        console.log(a + " " + b);
    }
}



